book/src/03_3_2_http_client.md

```diff
@@ -12,78 +10,63 @@ In this exercise, we'll write a small client that retrieves data over a HTTP con
 
 By default only unencrypted HTTP is available, which rather limits our options of hosts to connect to. We're going to use `http://neverssl.com/`.
 
-In `esp-idf`, HTTP client connections are managed by `http::client::EspHttpClient` in the `esp-idf-svc` crate. It implements the `http::client::Client` trait from `embedded-svc`, which defines functions for [HTTP request methods](https://en.wikipedia.org/wiki/Hypertext_Transfer_Protocol#Request_methods) like `GET` or `POST`.
+In `esp-idf`, HTTP client connections are managed by `http::client::EspHttpClient` in the `esp-idf-svc` crate. It implements the `http::client::Client` trait from `embedded-svc`, which defines functions for [HTTP request methods](https://en.wikipedia.org/wiki/Hypertext_Transfer_Protocol#Request_methods) like `GET` or `POST`. This is a good time to have a look at the documentation you opened with `cargo doc --open`.
+
+✅ Add the url `http://neverssl.com/` is in the main function. This is the address we will query.
 
-✅ Create a new `EspHttpClient` with default values.
+✅ Create a new `EspHttpClient` with default values (Try to find the method `new_default()` in the documentation).
 
-TODO currently no `docs.rs` documentation for the svc crates (ping Espressif about this, it's a known issue though)
 
-Calling HTTP functions (e.g. `get(url)`) on this client returns an `EspHttpRequest`, which must be turned into a `Writer` to reflect the client's option to send some data alongside its request. This makes more sense with `POST` and `PUT` but must still be performed with `GET`. 
+Calling HTTP functions (e.g. `get(url)`) on this client returns an `EspHttpRequest`, which must be turned into a `Writer` to reflect the client's option to send some data alongside its request. 
 
 After this optional send step the `Writer` can be turned into a `Response` from which the received server output can be read:
 
-TODO explain as_ref(). Or find a better way to point people to it. 
 
 ```Rust
-let request = client.get(url.as_ref())?;
-// the parameter passed to `into_writer` is the number of bytes
-// the client intends to send
+let request = client.get(url)?;
 let writer = request.into_writer(0)?;
 let response = writer.into_response()?;
 ```
-✅ Implement sending a `GET` request and receiving a response.
+The parameter passed to `into_writer` is the number of bytes the client intends to send. Here we are not trying to send anything. 
 
 A successful response has [a status code in the 2xx range](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes).
 
 ✅ Verify the connection was successful.
 
 ✅ Return an Error if the status is not in the 2xx range.
 
+```rust
+match status {
+        200..=299 => {
+        }
+        _ => anyhow::bail!("unexpected response code: {}", status),
+    }
+```
+The status error can be returned with [Anyhow](https://docs.rs/anyhow/latest/anyhow/index.html), who returns the result and manages internally any type `Error` that implements `std::error:Error`.
+
```

Comment:
I think here it would be better not to give away the solution as `match` is pretty straightforward.
```suggestion
Use `match` to ensure the status is in the 2xx range.  Return an error with [Anyhow](https://docs.rs/anyhow/latest/anyhow/index.html) if the status is outside of this range.